replication/README.md

```diff
@@ -350,3 +350,15 @@ message GetVolumeReplicationInfoResponse {
+| ------------------------------------------------ | --------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Missing required field                           | 3 INVALID_ARGUMENT    | Indicates that a required field is missing from the request.                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | Caller MUST fix the request by adding the missing required field before retrying.                                                                                                                                                     |
+| Volume does not exist or volume replication details not found                           | 5 NOT_FOUND           | Indicates that a volume corresponding to the specified `volume_id` does not exist or replication details are not avaiable at the moment.                                                                                                                                                                                                                                                                                                                                                                                                                            | Caller MUST verify that the `volume_id` is correct and that the volume is accessible and has not been deleted before retrying with exponential back off.                                                                              |
```

Comment:
"`volume_id` does not exist" should be FAILED_PRECONDITION and not NOT_FOUND.

Stating this as differentiating missing replication info and missing image would be difficult if they are the same error. The precondition error seems better as the image is missing, and to return status for the image the precondition is that it should be present (and other preconditions).

Reply:
volume_id is considered as a NOT_Found error for all RPC calls. IMO we should not treat it differently here.  If you still think it needs to be treated as `FAILED_PRECONDITION` good to open a separate issue and discuss over there?

Reply:
I guess we move the responsibility to the caller to check volume availability and retry, so this is fine for now.